File: capture/plugins/garland.c
/* garland.c
 *
 * Copyright 2025 Arkime. All rights reserved.
 *
 * SPDX-License-Identifier: Apache-2.0
 */

#include "arkime.h"
extern ArkimeConfig_t        config;

/******************************************************************************/
SUPPRESS_ALIGNMENT
LOCAL ArkimePacketRC arkime_packet_garland(ArkimePacketBatch_t *batch, ArkimePacket_t *const packet, const uint8_t *data, int len)
{
    if (len < 20)
        return ARKIME_PACKET_CORRUPT;

    return arkime_packet_run_ethernet_cb(batch, packet, data + 18, len - 18, ARKIME_ETHERTYPE_ETHER, "garland");
}
/******************************************************************************/
void arkime_plugin_init()
{
    uint32_t ethertype = arkime_config_int(NULL, "garlandEthertype", 0xff12, 0, 0xffff);
    arkime_packet_set_ethernet_cb(ethertype, arkime_packet_garland);
}

File: capture/plugins/garland.js
/******************************************************************************/
/* garland.js
 *
 * Copyright 2025 Arkime. All rights reserved.
 *
 * SPDX-License-Identifier: Apache-2.0
 */
'use strict';

let Pcap;

function garland (pcap, buffer, obj, pos) {
  return pcap.ethertyperun(0, buffer.slice(18), obj, pos + 18);
}

/// ///////////////////////////////////////////////////////////////////////////////
exports.init = function (config, emitter, api) {
  const Config = config;
  Pcap = api.getPcap();

  if (Pcap.setEtherCB === undefined) {
    console.error('ERROR - Garland plugin requires newer version of Arkime');
    return;
  }
  console.log('Garland plugin initialized');

  const ethertype = parseInt(Config.get('garlandEthertype', 0xff12));

  Pcap.setEtherCB(ethertype, garland);
};
